app/scripts/modules/titus/src/serverGroup/configure/wizard/pages/ServerGroupBasicSettings.tsx

```diff
@@ -0,0 +1,342 @@
+    const { setFieldValue, values } = this.props;
+    values.credentials = account;
+    values.credentialsChanged(values);
+    setFieldValue('credentials', account);
+  };
+
+  private regionUpdated = (region: string): void => {
+    const { values, setFieldValue } = this.props;
+    values.region = region;
+    values.regionChanged(values);
+    setFieldValue('region', region);
+  };
+
+  public validate(values: ITitusServerGroupCommand): { [key: string]: string } {
+    const errors: { [key: string]: string } = {};
+
+    if (!isStackPattern(values.stack)) {
+      errors.stack = 'Only dot(.) and underscore(_) special characters are allowed in the Stack field.';
+    }
+
+    if (!isDetailPattern(values.freeFormDetails)) {
+      errors.freeFormDetails =
+        'Only dot(.), underscore(_), and dash(-) special characters are allowed in the Detail field.';
+    }
+
+    if (!values.repository) {
+      errors.repository = 'Image is required.';
+    }
+    if (!values.tag) {
+      errors.tag = 'Tag is required.';
+    }
+
+    return errors;
+  }
+
+  private navigateToLatestServerGroup = () => {
+    const { values } = this.props;
+    const { latestServerGroup } = this.state;
+
+    const params = {
+      provider: values.selectedProvider,
+      accountId: latestServerGroup.account,
+      region: latestServerGroup.region,
+      serverGroup: latestServerGroup.name,
+    };
+
+    const { $state } = ReactInjector;
+    if ($state.is('home.applications.application.insight.clusters')) {
+      $state.go('.serverGroup', params);
+    } else {
+      $state.go('^.serverGroup', params);
+    }
+  };
+
+  private stackChanged = (stack: string) => {
+    this.props.setFieldValue('stack', stack);
+  };
+
+  private freeFormDetailsChanged = (freeFormDetails: string) => {
+    this.props.setFieldValue('freeFormDetails', freeFormDetails);
+  };
+
+  public componentWillReceiveProps(
+    nextProps: IServerGroupBasicSettingsProps & IWizardPageProps & FormikProps<ITitusServerGroupCommand>,
+  ) {
+    this.updateImageId(nextProps.values.repository, nextProps.values.tag);
+    this.setState(this.getStateFromProps(nextProps));
+  }
+
+  private strategyChanged = (values: ITitusServerGroupCommand, strategy: any) => {
+    values.onStrategyChange(values, strategy);
+    this.props.setFieldValue('strategy', strategy.key);
+  };
+
+  private dockerValuesChanged = (dockerValues: any) => {
+    Object.keys(dockerValues).forEach(key => {
+      this.props.setFieldValue(key, dockerValues[key]);
+    });
+  };
+
+  public render() {
+    const { errors, setFieldValue, values } = this.props;
+    const { createsNewCluster, latestServerGroup, namePreview, showPreviewAsWarning } = this.state;
+    const { AccountSelectField, DeploymentStrategySelector } = NgReact;
+
+    const accounts = values.backingData.accounts;
+    const readOnlyFields = values.viewState.readOnlyFields || {};
+
+    return (
+      <div className="container-fluid form-horizontal">
```

Comment:
Lots of JSX here.  Might be nice to split it up into smaller chunks.  This same comment applies to other components in this PR

Reply:
True, I'll take a look at breaking some of these things down a bit more. Just a straight conversion here, but can't hurt to improve things a bit in the process.